common/lib/capa/capa/tests/test_responsetypes.py

```diff
@@ -1292,6 +1321,26 @@ def test_grade_with_no_checkbox_selected(self):
         correct_map = problem.grade_answers({})
         self.assertEqual(correct_map.get_correctness('1_2_1'), 'incorrect')
 
+    def test_contextualized_choices(self):
+        script = textwrap.dedent("""
+            a = 6
+            b = 4
+            c = a + b
+
+            ok0 = c % 2 == 0 # check remainder modulo 2
+            ok1 = c % 3 == 0 # check remainder modulo 3
+            ok2 = c % 5 == 0 # check remainder modulo 5
+            ok3 = not any([ok0, ok1, ok2])
+        """)
+        choices = ["$ok0", "$ok1", "$ok2", "$ok3"]
+        problem = self.build_problem(script=script,
+                                     choice_type='checkbox',
+                                     choices=choices)
+
+        # Ensure the expected correctness
+        self.assert_grade(problem, ['choice_0', 'choice_2'], 'correct')
+        self.assert_grade(problem, ['choice_1', 'choice_3'], 'incorrect')
```

Comment:
Note: MultipleChoiceResponse problems allow you to [set a base `name` attribute for each choice](https://github.com/open-craft/edx-platform/blob/2537303fcdecd574dc2463b60ee71fb482d23a8f/common/lib/capa/capa/responsetypes.py#L1256-L1260), however [ChoiceResponse problems don't](https://github.com/open-craft/edx-platform/blob/2537303fcdecd574dc2463b60ee71fb482d23a8f/common/lib/capa/capa/responsetypes.py#L870-L880).

So that's why this test uses `choice_0`, `choice_1` ..., and the [above test uses custom choice name values](https://github.com/edx/edx-platform/pull/13534/files#diff-eb476e271b5d55f9717fb3137459752aR197).